### PR TITLE
[TGUI] Coin Mint

### PR DESCRIPTION
**Pull request by AyIong**
## What Does This PR Do
Coin Mint is now on TGUI, very compact!

Refactored Coin Mint. 
- Added ability to remove inserted materials. 
- Removed custom print count, now it will create coins until materials gone or money bag be full, btw... 
- MONEY BAG! It will no longer appear out of nowhere, and also Coin Mint now requires a money bag to be inserted in it to work.
- Coins are now creating 4 times slower (2 seconds - every SS update).

The money bag can now be crafted from 3 cloths and storage slots increaced from 40 to 50.

For reviewers: please don't kill me....

## Why It's Good For The Game
First of all, **the** TGUI
Secondly, more logic in behaviour
Thirdly, money bags don't appear from nothing

Closes - https://github.com/orgs/ParadiseSS13/projects/4?pane=issue&itemId=38641487

## Images of changes
I love comparisons

| Before | After |
| - | - |
| image | image |

## Testing
Yes, and i'm tired...
<details>
<summary>Video</summary>

https://github.com/ParadiseSS13/Paradise/assets/69762909/f93a66ed-5c0f-4f3a-acc7-e2ba6153f970

</details>

## Changelog
:cl:
add: Coin Mint now have TGUI.
tweak: Coin Mint now requires a money bag and won't work without it, also coins are now produced 4 times slower.
tweak: Now you can make a money bag out of 3 cloths.
/:cl:

### CODE REVIEW

File: code/modules/mining/mint.dm [hunk 1]
```diff
@@ -1,116 +1,193 @@
...
+			if(isnull(num_sheets))
+				return
+			materials.retrieve_sheets(num_sheets, chosen_material)
+		if("ejectBag")
+			eject_bag(usr)
+
+/obj/machinery/mineral/mint/attackby(obj/item/I, mob/user, params)
+	if(istype(I, /obj/item/storage/bag/money))
+		if(money_bag)
+			to_chat(user, "<span class='notice'>There is already [money_bag.name] inside!</span>")
```
Comment (Henri215) [inline code/modules/mining/mint.dm:128]: "There is already money bag inside!" Missing an "a" here.

Comment (AyIong) [inline code/modules/mining/mint.dm:128]:
```suggestion
			to_chat(user, "<span class='notice'>There is already a [money_bag.name] inside!</span>")
```

File: code/modules/mining/mint.dm [hunk 2]
```diff
@@ -1,116 +1,193 @@
...
+			eject_bag(usr)
+
+/obj/machinery/mineral/mint/attackby(obj/item/I, mob/user, params)
+	if(istype(I, /obj/item/storage/bag/money))
+		if(money_bag)
+			to_chat(user, "<span class='notice'>There is already a [money_bag.name] inside!</span>")
+			return
+		if(!user.drop_item())
+			return
+		to_chat(user, "<span class='notice'>You put a [I.name] into a [src] </span>")
```
Comment (Henri215) [inline code/modules/mining/mint.dm:132]:
```suggestion
		to_chat(user, "<span class='notice'>You put a [I.name] into a [src].</span>")
```

**Review by S34NW — APPROVED** [review]

**Review by Burzah — APPROVED** [review]

File: code/modules/mining/mint.dm [hunk 3]
```diff
@@ -1,116 +1,193 @@
...
-	AddComponent(/datum/component/material_container, list(MAT_METAL, MAT_PLASMA, MAT_SILVER, MAT_GOLD, MAT_URANIUM, MAT_DIAMOND, MAT_BANANIUM, MAT_TRANQUILLITE), MINERAL_MATERIAL_AMOUNT * 50, FALSE, /obj/item/stack)
+	var/static/list/coin_materials = list()
+	if(!length(coin_materials))
+		for(var/datum/material/coin_mat as anything in subtypesof(/datum/material))
+			var/obj/item/coin/coin_type = coin_mat.coin_type
+			if(!coin_type)
+				continue
+			coin_materials += coin_mat.id
+
+	AddComponent(/datum/component/material_container, coin_materials, MINERAL_MATERIAL_AMOUNT * 50, FALSE, /obj/item/stack, _after_insert=CALLBACK(src, PROC_REF(material_insert)))
```
Comment (Henri215) [inline code/modules/mining/mint.dm:28]:
```suggestion
	AddComponent(/datum/component/material_container, coin_materials, MINERAL_MATERIAL_AMOUNT * 50, FALSE, /obj/item/stack, _after_insert = CALLBACK(src, PROC_REF(material_insert)))
```

File: code/modules/mining/mint.dm [hunk 4]
```diff
@@ -1,116 +1,193 @@
...
+	chosen_material = pick(coin_materials[1])
+
+/obj/machinery/autolathe/Destroy()
```
Comment (Henri215) [inline code/modules/mining/mint.dm:31]:
```suggestion
/obj/machinery/mineral/mint/Destroy()
```
Looks like you forgot to change this when copy and pasting.

**Review by Henri215 — CHANGES_REQUESTED** [review]

File: code/modules/mining/mint.dm (hunk 4 above)
Comment (AyIong) [reply code/modules/mining/mint.dm:31]: Looks like yes

**Review by Henri215 — APPROVED** [review]
image

**Comment by AyIong** [comment]
> image

money-sniff

**Pull request merged**